idea/src/org/jetbrains/kotlin/idea/inspections/ConstantConditionIfInspection.kt

```diff
@@ -116,12 +119,47 @@ class ConstantConditionIfInspection : AbstractKotlinInspection() {
 private fun KtIfExpression.branch(thenBranch: Boolean) = if (thenBranch) then else `else`
 
 private fun KtExpression.constantBooleanValue(context: BindingContext): Boolean? {
+    val enumEntriesComparison = enumEntriesComparison()
+    if (enumEntriesComparison != null) {
+        return enumEntriesComparison
+    }
     val type = getType(context) ?: return null
-
     val constantValue = ConstantExpressionEvaluator.getConstant(this, context)?.toConstantValue(type)
     return constantValue?.value as? Boolean
 }
 
+private fun KtExpression.enumEntriesComparison(): Boolean? {
+    if (this !is KtBinaryExpression) return null
+    val left = this.left as? KtDotQualifiedExpression ?: return null
+    val right = this.right as? KtDotQualifiedExpression ?: return null
+
+    val leftEnum = left.enumClass() ?: return null
+    val rightEnum = right.enumClass() ?: return null
+    if (leftEnum != rightEnum) return null
```

Comment:
It's correct, because for entries of different enum classes we have the compiler warning `INCOMPATIBLE_ENUM_COMPARISON`